src/adapters/beachfront.js

```diff
@@ -0,0 +1,129 @@
+import Adapter from 'src/adapters/adapter';
+import bidfactory from 'src/bidfactory';
+import bidmanager from 'src/bidmanager';
+import * as utils from 'src/utils';
+import { ajax } from 'src/ajax';
+import { STATUS } from 'src/constants';
+
+const ENDPOINT = '//reachms.bfmio.com/bid.json?exchange_id=';
+
+function BeachfrontAdapter() {
+  var baseAdapter = Adapter.createNew('beachfront');
+
+  baseAdapter.callBids = function (bidRequests) {
+    const bids = bidRequests.bids || [];
+    bids.forEach(function(bid) {
+      var bidRequest = getBidRequest(bid);
+      var RTBDataParams = prepareAndSaveRTBRequestParams(bid);
+      if (!RTBDataParams) {
+        var error = "No bid params";
+        utils.logError(error);
+      }
+      var BID_URL = ENDPOINT + RTBDataParams.appId;
+      ajax(BID_URL, handleResponse(bidRequest), JSON.stringify(RTBDataParams), {
+        contentType: 'text/plain',
+        withCredentials: true
+      });
+    });
+
+  };
+
+  function getBidRequest(bid) {
+    if (!bid || !bid.params || !bid.params.appId) {
+      return;
+    }
+
+    var bidRequest = bid;
+    bidRequest.width = parseInt(bid.sizes[0], 10) || undefined;
+    bidRequest.height = parseInt(bid.sizes[1], 10) || undefined;
+    return bidRequest;
+  }
+
+  function prepareAndSaveRTBRequestParams(bid) {
+    if (!bid || !bid.params || !bid.params.appId) {
```

Comment:
Your documentation says `bid.params.bidfloor` is also required, so that should probably be here.